### PR TITLE
jruby support

### PR DESCRIPTION
**Pull request by ahorek**
this PR imports a Java version of the CGI library, see https://github.com/jruby/jruby/issues/6682

JRuby contribs
https://github.com/jruby/jruby/commits/master/core/src/main/java/org/jruby/ext/cgi/escape/CGIEscape.java

Charles Oliver Nutter (@headius)
Thomas Enebo (@enebo)

could you grant permission to relicense the code to BSD 2-Clause "Simplified" License ?

### CODE REVIEW

**Comment by headius** [comment]
I consent to having my contributions to the JRuby native cgi/escape library relicensed under the terms of the BSD 2-Clause License and the Ruby license.

**Comment by enebo** [comment]
I consent to having my contributions to the JRuby native cgi/escape library relicensed under the terms of the BSD 2-Clause License and the Ruby license.

File: cgi.gemspec [hunk 1]
```diff
@@ -22,10 +22,25 @@ Gem::Specification.new do |spec|
...
 
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z 2>/dev/null`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
-  end
   spec.bindir        = "exe"
   spec.executables   = []
+
+  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
+    `git ls-files -z 2>/dev/null`.split("\x0").reject { |f| f.match(%r{^(test|spec|features|.github)/}) }
+  end
```
Comment (olleolleolle) [inline cgi.gemspec:30]: What if this is copy-pasta?

```suggestion
```

Comment (ahorek) [reply cgi.gemspec]: it is! thanks

**Comment by headius** [comment]
How are we looking here?

**Comment by hsbt** [comment]
@headius @ahorek Basically, I'm +1 to merge this. But I don't have to coordinate the directory structure same as https://github.com/ruby/digest/pull/25

**Comment by ahorek** [comment]
@hsbt I rebased the PR against the current master, please let me know if there's any misplaced file, but the directory structure should match our previous discussion about the digest gem. In short, all JRuby-specific files are separated to ```cgi/ext/java/*``` and they're not included in the CRuby version of the gem.

**Comment by headius** [comment]
@hsbt I don't think this has made it into a release yet... can we make that happen?

**Comment by hsbt** [comment]
@headius Oh sorry. I released v0.3.3 with these changes.

**Comment by headius** [comment]
@hsbt Thank you!

**Pull request merged**